pyi.py

```diff
@@ -1656,8 +1656,7 @@ def run(self) -> Iterable[Error]:
         if path.suffix == ".pyi":
             yield from _check_for_type_comments(path)
             visitor = PyiVisitor(filename=path)
-            for error in visitor.run(LegacyNormalizer().visit(self.tree)):
-                yield error
+            yield from visitor.run(LegacyNormalizer().visit(self.tree))
```

Comment:
Instead of doing this:

```python
visitor = PyiVisitor(filename=path)
yield from visitor.run(LegacyNormalizer().visit(self.tree))
```

I would find it more readable if we did this:

```python
tree = LegacyNormalizer().visit(self.tree)
yield from PyiVisitor(filename=path).run(tree)
```